addons/binding/org.openhab.binding.rfxcom/src/main/java/org/openhab/binding/rfxcom/internal/messages/RFXComCurrentEnergyMessage.java

```diff
@@ -0,0 +1,225 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ * <p>
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.rfxcom.internal.messages;
+
+import org.eclipse.smarthome.core.library.items.NumberItem;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.Type;
+import org.openhab.binding.rfxcom.RFXComValueSelector;
+import org.openhab.binding.rfxcom.internal.exceptions.RFXComException;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * RFXCOM data class for Current and Energy message.
+ *
+ * @author Damien Servant
+ * @since 1.9.0
+ */
+public class RFXComCurrentEnergyMessage extends RFXComBaseMessage {
+    private static float TOTAL_USAGE_CONVERSION_FACTOR = 223.666F;
+
+    public enum SubType {
+        ELEC4(1), // OWL - CM180i
+
+        UNKNOWN(255);
+
+        private final int subType;
+
+        SubType(int subType) {
+            this.subType = subType;
+        }
+
+        public byte toByte() {
+            return (byte) subType;
+        }
+
+        public static SubType fromByte(int input) {
+            for (SubType c : SubType.values()) {
+                if (c.subType == input) {
+                    return c;
+                }
+            }
+
+            return SubType.UNKNOWN;
+        }
+    }
+
+    private final static List<RFXComValueSelector> supportedInputValueSelectors = Arrays.asList(
+            RFXComValueSelector.SIGNAL_LEVEL, RFXComValueSelector.BATTERY_LEVEL, RFXComValueSelector.CHANNEL1_AMPS,
+            RFXComValueSelector.CHANNEL2_AMPS, RFXComValueSelector.CHANNEL3_AMPS, RFXComValueSelector.TOTAL_USAGE);
+
+    private final static List<RFXComValueSelector> supportedOutputValueSelectors = Arrays.asList();
+
+    public SubType subType = SubType.UNKNOWN;
+    public int sensorId = 0;
+    public byte count = 0;
+    public double channel1Amps = 0.0;
+    public double channel2Amps = 0.0;
+    public double channel3Amps = 0.0;
+    public double totalUsage = 0.0;
+    public byte signalLevel = 0;
+    public byte batteryLevel = 0;
+
+    public RFXComCurrentEnergyMessage() {
+        packetType = PacketType.CURRENT_ENERGY;
+    }
+
+    public RFXComCurrentEnergyMessage(byte[] data) {
+        encodeMessage(data);
+    }
+
+    @Override
+    public String toString() {
+        String str = "";
+
+        str += super.toString();
+        str += "\n - Sub type = " + subType;
```

Comment:
Please, remove all new lines from generated string (replace by comma), so that debug printouts are inline with other rfxcom messages in OH2